### PR TITLE
Removes month and year from activity_info table

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/3407
Resolves https://github.com/brave/brave-browser/issues/2911
Resolves https://github.com/brave/brave-browser/issues/2943

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
- enable rewards
- visit site and make sure that you see site in the panel
- make sure that it's shown in ac table

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [x] Adequate test coverage exists to prevent regressions 
- [x] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: components/brave_rewards/browser/publisher_info_database.cc [hunk 1]
```diff
@@ -1106,6 +1089,99 @@ bool PublisherInfoDatabase::MigrateV4toV5() {
...
+               "SELECT " + columns_select + " "
+               "FROM activity_info_old "
+               "GROUP BY publisher_id, reconcile_stamp;");
+    sql.append("DROP TABLE activity_info_old;");
+    sql.append("PRAGMA foreign_keys=on;");
+
+    bool result = GetDB().Execute(sql.c_str());
+
+    if (!result) {
+      LOG(ERROR) << "DB: Error with MigrateV5toV6";
```
Comment (tmancey) [inline components/brave_rewards/browser/publisher_info_database.cc]: If there is an error should we not `return false`?

File: components/brave_rewards/browser/publisher_info_database.cc [hunk 2]
```diff
@@ -1117,33 +1193,10 @@ sql::InitStatus PublisherInfoDatabase::EnsureCurrentVersion() {
...
-  // to version 5
-  if (old_version < 5 && cur_version < 6) {
-    if (!MigrateV4toV5()) {
-      LOG(ERROR) << "DB: Error with MigrateV4toV5";
-    }
+  for (auto i = start_version; i <= cur_version; i++) {
+    Migrate(i);
   }
 
   meta_table_.SetVersionNumber(cur_version);
```
Comment (tmancey) [inline components/brave_rewards/browser/publisher_info_database.cc]: nit: cur_version and CurrentVersion, would current_version be better for consistency

File: components/brave_rewards/browser/publisher_info_database.cc (hunk 1 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/publisher_info_database.cc]: we have this queries in transaction, so transaction at the end will fail

Comment (NejcZdovc) [reply components/brave_rewards/browser/publisher_info_database.cc]: what would you suggest to do in this case?

Comment (NejcZdovc) [reply components/brave_rewards/browser/publisher_info_database.cc]: we have all of this code wrapped in transaction, so transaction will fail. I just added this log as I needed to used return value from Execute. But now that you pointed out I think we shouldn't increase db number if migration fail

File: components/brave_rewards/browser/publisher_info_database.cc (hunk 2 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/publisher_info_database.cc]: done

File: components/brave_rewards/browser/publisher_info_database.cc (hunk 1 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/publisher_info_database.cc]: done

File: components/brave_rewards/browser/publisher_info_database.h [hunk 3]
```diff
@@ -115,6 +119,7 @@ class PublisherInfoDatabase {
 
   sql::MetaTable& GetMetaTable();
 
+
```
Comment (jasonrsadler) [inline components/brave_rewards/browser/publisher_info_database.h]: Think linter will fail on 2 blank lines

File: vendor/bat-native-ledger/src/bat_publishers.cc [hunk 4]
```diff
@@ -140,12 +130,8 @@ ledger::ActivityInfoFilter BatPublishers::CreateActivityFilter(
 
 ledger::ActivityInfoFilter BatPublishers::CreateActivityFilter(
```
Comment (jasonrsadler) [inline vendor/bat-native-ledger/src/bat_publishers.cc]: Before, we were keying from pub_id, month, and year (which normally would return one entry). It seems like some of these filters just key off of pub_id which can return multiples over several contribution months. Do we want to have reconcile_stamp required for these?

**Comment by jasonrsadler** [comment]
Need test plan

File: components/brave_rewards/browser/publisher_info_database.h (hunk 3 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/publisher_info_database.h]: done

File: vendor/bat-native-ledger/src/bat_publishers.cc (hunk 4 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat_publishers.cc]: removed unused filters

**Comment by NejcZdovc** [comment]
@jasonrsadler test plan added

**Pull request merged**